### ISSUE TITLE
Governor using MultiTokens

### ISSUE THREAD

**Issue by EmanuelCampos** [OpenZeppelin/openzeppelin-contracts#3423]
**🧐 Motivation**
It's a feature that will provide an Extension to use more than one token with Governor.

**📝 Details**
- [] need to be an extension
- [] need to accept a list of tokens
- [] need to provide support to governance and getPastVotes/ getPastTotalSupply
-

**Comment by EmanuelCampos** [comment]
@frangio could you tell me why this approach doesn't work? could you give me suggestions to improve it?

**Comment by frangio** [comment]
Thanks for the suggestion. Before submitting a PR, can you provide some more context? What is the use case that made you interested in this feature?

If we implement this, it should probably be a Governor module similar to [GovernorVotes](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/governance/extensions/GovernorVotes.sol). The important part is implementing the `_getVotes` function.

**Comment by EmanuelCampos** [comment]
@frangio it will be a Governance with 2 tokens, to two different people, like staking tokens and the common token, so who have the staking token have the governance power too, and it can grow more to other tokens

**Comment by EmanuelCampos** [comment]
@frangio I was using the getPastVotes, is it much different from my Implementation?

**Comment by frangio** [comment]
Yes, it's very different. All that's needed is `_getVotes`. Please read the code and make sure you understand that.

I'm concerned about the storage overhead that it will imply for castVote. I think this feature should wait until there are immutable arrays (https://github.com/ethereum/solidity/issues/12587), which are allegedly being worked on.

I would put this feature request on hold, but I encourage you to implement it in your project for 2 tokens based on GovernorVotes.

**Comment by Amxx** [comment]
@EmanuelCampos 
Relevant piece of code: https://github.com/p00ls/contracts/blob/master/contracts/dao/P00lsDAO.sol#L41-L52

**Comment by EmanuelCampos** [comment]
@Amxx @frangio aI believe this both approaches using only `getVotes` will not provide support to https://github.com/OpenZeppelin/openzeppelin-contracts/blob/65b45726b34dafe8fc3ef78c3d4b7b3f404f61ad/contracts/governance/extensions/GovernorVotesQuorumFraction.sol#L47-L49, so I did my implementation thinking on this case

**Comment by frangio** [comment]
I see. It's true that the quorum module becomes incompatible. With the approach I suggested I would include the quorum logic along with the custom `_getVotes` implementation. I'm not sure that deploying a separate contract to wrap the multiple tokens is the right way to implement this.

**Comment by bilalmotiwala** [comment]
Hey folks! Just sort of checking in on this. Was anyone able to actually come up with a way to use multiple tokens for their governance?

**Comment by Amxx** [comment]
> Hey folks! Just sort of checking in on this. Was anyone able to actually come up with a way to use multiple tokens for their governance?

If all your tokens are voting tokens, you can follow this approach https://github.com/p00ls/contracts/blob/master/contracts/dao/P00lsDAO.sol#L41-L52
- override `_getVotes` 
  - call `getPastVotes` for all your contracts
  - sum that up
  - return that "aggregated" vote count.
